src/Renderer/Swapchain/Swapchain.h

```diff
@@ -4,58 +4,212 @@
 
 namespace SnekVk
 {
+    /**
+     * @brief The Swapchain class defines the creation and usage of a graphics swapchain. 
+     * Swapchains define how images are rendered to screen. Generally, rendered objects are
+     * drawn to images, which are contained within command buffers. Each frame, an image is drawn
+     * to a command buffer which is then presented for rendering.
+     * 
+     * When an image takes longer than the given time for a frame, it runs the risk of overlapping
+     * with the image being written in the following frame. This often results in screen tearing, 
+     * where two images are overlain on top of each other. 
+     * 
+     * The swapchain is therefore responsible for buffering these images by writing to a new commandbuffer 
+     * each frame. The new commandbuffer is then swapped out each frame depending on the method used. This
+     * results in smooth image transitions and safer memory usage, as frames no longer compete to render the same data. 
+     * 
+     * Swapchains are essential for computer graphics processing. 
+     */
     class SwapChain
     {
         public:
 
+        /** 
+         *  When swapping out images for our frames, we can have multiple frames 'in flight', 
+         *  meaning frames that act as additional memory buffers. In our case we allow for two additional
+         *  frames in flignt for added image buffering. 
+         **/
         static constexpr int MAX_FRAMES_IN_FLIGHT = 2;
 
         // 'Structors 
 
+        /**
+         * @brief Creates a new Swapchain object to buffer our images. 
+         * 
+         * @param device a reference to a device. Devices are always destroyed last, so we can 
+         * guarantee that object destruction will occur at the correct time. 
+         * @param windowExtent the window resolution. Images need to be scaled to an extent, this usually
+         * must follow the extents of the window. 
+         */
         SwapChain(VulkanDevice& device, VkExtent2D windowExtent);
```

Comment:
I remember you said that you'd eventually remove references to devices, so I think this is changed later.